src/LinAlg/hiopMatrixSparseTriplet.cpp

```diff
@@ -141,14 +141,45 @@ void hiopMatrixSparseTriplet::addSubDiagonal(const double& alpha, long long star
   assert(false && "not needed");
 }
 
+void hiopMatrixSparseTriplet::copySubDiagonalEleFromVec(const long long& start_on_dest_diag, const long long& num_elems,
```

Comment:
I think this kernel needs a unit test.

Reply:
add a unit test into test/testMatrixSparse.cpp.
The corresponding code is added into test/LinAlg/matrixTestsSparseTriplet.cpp
However, the corresponding function verifyAnser(...) hasn't been implemented. Will set a follow-up meeting with  Asher @ashermancinelli to discuss this issue.

Reply:
Update: @nychiang and I discussed our unit test conventions, so he should now be able to update accordingly.